src/App.tsx

```diff
@@ -15,48 +22,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreparedGoods(
+  goods: string[],
+  sortType: Buttons,
+  isReversed: boolean,
+) {
+  const copyGoods = [...goods];
+
+  if (sortType) {
+    copyGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case Buttons.SORT__ALPHBETICALLY:
+          return good1.localeCompare(good2);
+        case Buttons.SORT__BY__LENGTH:
+          return good1.length - good2.length;
+
+        default: return 0;
+      }
+    });
+  }
+
+  return isReversed ? copyGoods.reverse() : copyGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setsortType] = useState<Buttons>(Buttons.SORT__BY__DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+  const preparedGoods = getPreparedGoods(
+    goodsFromServer, sortType, isReversed,
+  );
+
+  const handleReset = () => {
+    setsortType(Buttons.SORT__BY__DEFAULT);
+    setIsReversed(false);
+  };
+
+  const resetCondition = (
```

Comment:
```suggestion
  const isResetVisible = (
```
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/96409563/abce980f-b1ee-4db0-a98f-ff535b47b8d8)